src/mach/load_command.rs

```diff
@@ -371,44 +371,137 @@ pub const SIZEOF_DYLINKER_COMMAND: usize = 12;
 // struct XXX_thread_state state   thread state for this flavor
 // ...
 #[repr(C)]
-#[derive(Debug, Clone, Copy, Pread, Pwrite, SizeWith)]
-pub struct ThreadCommand {
+#[derive(Debug, Copy, Clone)]
+pub struct ThreadCommand<'a> {
```

Comment:
I saw these constraints:

1. `impl TryFromCtx for ThreadCommand` doesn't know `cputype`
2. `ThreadCommand` needs to be `Copy`
3. A `pwrite` implementation for `ThreadCommand` should be able to reproduce arbitrary thread states

A `&[u8]` fit the bill, but I figured it would be controversial. If we relax the last constraint from "arbitrary thread states" to "thread states corresponding to one of these four structs", there's a lot of other options.

* We could do some type level stuff to make `ThreadCommand` generic over `cputype`. That type parameter would have to bubble up to `LoadCommand`, which complicates things, but it would mean that `ThreadCommand` could directly contain a different machine-specific struct describing the register state.
* We could treat machine-specific thread states as an untagged union. That avoids the type parameter, but it bumps the minimum to Rust 1.19. It also wouldn't work for big-endian host machines, so nevermind.
* We could treat machine-specific thread states as a series of separate structs, and just… read into all of them. That grows `ThreadCommand` to ~800 bytes and complicates writing (which struct do we write?) but it would solve the reading problem.
* We could treat thread state as an opaque array, say a `[u32; 70]`, and worry about extracting meaningful data ourselves.

Having written all that out, the `[u32; 70]` approach seems least invasive and has the fewest downsides, so I'll go with that.